x-pack/plugins/apm/server/routes/mobile/get_mobile_launches_by_location.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  kqlQuery,
+  rangeQuery,
+  termQuery,
+} from '@kbn/observability-plugin/server';
+import {
+  LABEL_LIFECYCLE_STATE,
+  SERVICE_NAME,
+} from '../../../common/es_fields/apm';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { getOffsetInMs } from '../../../common/utils/get_offset_in_ms';
+import { getBucketSize } from '../../../common/utils/get_bucket_size';
+import { environmentQuery } from '../../../common/utils/environment_query';
+
+interface Props {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  locationField?: string;
+  offset?: string;
+}
+
+export async function getLaunchesByLocation({
+  kuery,
+  apmEventClient,
+  serviceName,
+  environment,
+  start,
+  end,
+  locationField,
+  offset,
+}: Props) {
+  const { startWithOffset, endWithOffset } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const { intervalString } = getBucketSize({
+    start: startWithOffset,
+    end: endWithOffset,
+    minBucketSize: 60,
+  });
+
+  const aggs = {
+    launches: {
+      filter: {
+        terms: { [LABEL_LIFECYCLE_STATE]: ['created', 'active'] },
+      },
+      aggs: {
+        byLocation: {
+          terms: {
+            field: locationField,
+          },
+        },
+      },
+    },
+  };
+
+  const response = await apmEventClient.logEventSearch(
```

Comment:
It could be just me but I find it easier to read the search query when I have access to the indices without needing to refer to the `logEventSearch` specification. Additionally, this makes `logEventSearch` a bit more abstract. WDYT?

```
 const response = await apmEventClient.logEventSearch(
    'get_mobile_location_launches',
    {
      apm: {
          events: [ProcessorEvent.error],
      },
      body: {
        track_total_hits: false,
        size: 0,
        query: ....
    }
```

Reply:
Oh got it! The reason I didn't include the option to pass indices to the function is because there's only one type of index needed for this use case, which is `ProcessorEvent.error`, and since other `ProcessorEvent` values won't work, I think it might be confusing, for someone not familiar with this function, that they can't pass other value. Also, tbh the `ProcessorEvent.error` indices do work for this use case, but it's out of coincidence really, because the logs we're querying here aren't errors, so I think that in the future, probably a new enum value along with a new config param would have to be added to set indices for plain logs only, and in that case I think it should be easier to keep the index decision inside the function so that its callers won't be affected by any changes regarding indices later.

Having said that though, I'm not aware of TS programming conventions, so if you think that passing the`ProcessorEvent.error` value for this use case would make it more friendly to use in general, then I'm ok to make the changes 👍

Reply:
>, so I think that in the future, probably a new enum value along with a new config param would have to be added to set indices for plain logs only, and in that case I think it should be easier to keep the index decision inside the function so that its callers won't be affected by any changes regarding indices later.

that's a really good point, let's leave it as is and we can revisit it if needed.